internal/webhook/external/quay_test.go

```diff
@@ -0,0 +1,186 @@
+				return fake.NewClientBuilder().
+					WithScheme(scheme).
+					WithObjects(
+						&corev1.Secret{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "fakesecret",
+								Namespace: "fakenamespace",
+							},
+							Data: map[string][]byte{
+								kargoapi.WebhookReceiverSecretKeyQuay: []byte("mysupersecrettoken"),
+							},
+						},
+						&kargoapi.ProjectConfig{
+							ObjectMeta: metav1.ObjectMeta{
+								Namespace: "fakenamespace",
+								Name:      "fakename",
+							},
+							Spec: kargoapi.ProjectConfigSpec{
+								WebhookReceivers: []kargoapi.WebhookReceiverConfig{
+									{
+										Quay: &kargoapi.QuayWebhookReceiver{
+											SecretRef: corev1.LocalObjectReference{
+												Name: "fakesecret",
+											},
+										},
+									},
+								},
+							},
+							Status: kargoapi.ProjectConfigStatus{
+								WebhookReceivers: []kargoapi.WebhookReceiver{
+									{
+										Path: GenerateWebhookPath(
+											"fake-webhook-receiver-name",
+											"fakename",
+											kargoapi.WebhookReceiverTypeQuay,
+											"mysupersecrettoken",
+										),
+									},
+								},
+							},
+						},
+						&kargoapi.Warehouse{
+							ObjectMeta: metav1.ObjectMeta{
+								Namespace: "fakenamespace",
+								Name:      "fakename",
+							},
+							Spec: kargoapi.WarehouseSpec{
+								Subscriptions: []kargoapi.RepoSubscription{
+									{
+										Image: &kargoapi.ImageSubscription{
+											RepoURL: "quay.io/mynamespace/repository",
+										},
+									},
+								},
+							},
+						},
+					).
+					WithIndex(
+						&kargoapi.Warehouse{},
+						indexer.WarehousesBySubscribedURLsField,
+						indexer.WarehousesBySubscribedURLs,
+					).
+					WithIndex(
+						&kargoapi.ProjectConfig{},
+						indexer.ProjectConfigsByWebhookReceiverPathsField,
+						indexer.ProjectConfigsByWebhookReceiverPaths,
+					).
+					Build()
+			},
+			req: func() *http.Request {
+				b := newQuayPayload()
+				req := httptest.NewRequest(http.MethodPost, url, b)
+				return req
+			},
+			secret: "fakesecret",
+			msg:    "{\"msg\":\"refreshed 1 warehouse(s)\"}\n",
+			code:   http.StatusOK,
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			req := test.req()
+			l := logging.NewLogger(logging.DebugLevel)
+			ctx := logging.ContextWithLogger(req.Context(), l)
+			req = req.WithContext(ctx)
+			w := httptest.NewRecorder()
+			namespace := "fakenamespace"
+			h := quayHandler(
+				test.kClient(),
+				namespace,
+			)
+			h(w, req)
+			require.Equal(t, test.code, w.Code)
+			require.Contains(t, w.Body.String(), test.msg)
+		})
+	}
+}
+
+func newQuayPayload() *bytes.Buffer {
+	return bytes.NewBuffer([]byte(`
+		{
+			"name": "repository",
+			"repository": "mynamespace/repository",
+			"namespace": "mynamespace",
+			"docker_url": "quay.io/mynamespace/repository",
+			"homepage": "https://quay.io/repository/mynamespace/repository",
+			"updated_tags": [
+			  "latest"
+			]
+		  }
```

Comment:
would it make sense to simplify the payload like this, assuming we don’t currently need the other fields for our tests? if, I'd also trim down the docker hub test payloads to only include the required fields
but that said, if we anticipate needing fields like name, repository, namespace, or updated_tags in the near future (for validation or richer logging) then it might be worth keeping them in place. wdyt

```suggestion
		{
			"docker_url": "quay.io/mynamespace/repository"
		  }
```